beacon-chain/core/blocks/exit.go

```diff
@@ -61,8 +63,15 @@ func ProcessVoluntaryExits(
 		if err := VerifyExitAndSignature(val, beaconState.Slot(), beaconState.Fork(), exit, beaconState.GenesisValidatorsRoot()); err != nil {
 			return nil, errors.Wrapf(err, "could not verify exit %d", idx)
 		}
-		beaconState, err = v.InitiateValidatorExit(ctx, beaconState, exit.Exit.ValidatorIndex)
-		if err != nil {
+		beaconState, exitEpoch, err = v.InitiateValidatorExit(ctx, beaconState, exit.Exit.ValidatorIndex, maxExitEpoch, churn)
+		if err == nil {
+			if exitEpoch > maxExitEpoch {
+				maxExitEpoch = exitEpoch
+				churn = 1
+			} else if exitEpoch == maxExitEpoch {
+				churn++
+			}
+		} else if err != v.ValidatorAlreadyExitedErr {
```

Comment:
I am not sure that I like silently doing nothing when a validator already exited. We could at least log something.

Reply:
This is the current behavior, there's many reasons to process a voluntary exit of an already exited validator, for example if the validator is slashed after exited, we would call this function anyway.

Reply:
can we use the `errors.Is` pattern here to check for particular error ? Same for the other cases